lxd/devices.go

```diff
@@ -379,6 +420,25 @@ func deviceTaskBalance(s *state.State) {
 	balancedInstances := map[instance.Instance]int{}
 	for _, c := range instances {
 		conf := c.ExpandedConfig()
+
+		cpuNodes, cpuNodesEnabled := conf["limits.cpu.nodes"]
+		if cpuNodesEnabled && cpuNodes == "" {
```

Comment:
This line didn't make sense to me.

In the LXD configuration format, an empty string is the same as not being set, so really this block isn't needed and instead line 430 can just be:

```go
cpuNodes := conf["limits.cpu.nodes"]
if cpuNodes != "" {}
```

Then on line 542, can we use `if len(numaCpus) > 0 {}` instead?